astronomer/airflow/version_check/update_checks.py

```diff
@@ -338,56 +350,130 @@ def __init__(self):
             static_folder='static',
             template_folder=os.path.join(os.path.dirname(__file__), "templates"),
         )
+        from .plugin import dismissal_period_days, eol_warning_threshold_days
 
-    def available_update(self):
-        from .models import AstronomerAvailableVersion
-
-        session = get_state(app=current_app).db.session
-        available_releases = session.query(AstronomerAvailableVersion).filter(
-            AstronomerAvailableVersion.hidden_from_ui.is_(False)
-        )
+        self.eol_warning_threshold_days = eol_warning_threshold_days
+        self.dismissal_period_days = dismissal_period_days
 
-        runtime_version = version.parse(get_runtime_version())
-        base_version = runtime_version.major
+    def get_update_notice(
+        self, sorted_releases: list, runtime_version: version.Version, base_version: int
+    ) -> dict[str, Any] | None:
+        """
+        Get the update notice information if there is a newer version available.
 
-        sorted_releases = sorted(available_releases, key=lambda v: version.parse(v.version), reverse=True)
+        :param sorted_releases: A list of available releases sorted by version.
+        :param runtime_version: The current runtime version.
+        :param base_version: The base version of the runtime.
+        """
         for rel in sorted_releases:
             # Only notify about the latest release if the user is in the highest patch level.
             # On runtime:
             # if the user is on version 5.0.6 and 5.0.8, 6.0.0 are available,
             # notify the user about 5.0.8 and don't notify user about 6.0.0.
             rel_parsed_version = version.parse(rel.version)
-
             rel_parsed_base_version = rel_parsed_version.major
 
             if rel_parsed_version > runtime_version and rel_parsed_base_version == base_version:
+                days_to_eol = None
+                if rel.end_of_support:
+                    now = utcnow()
+                    days_to_eol = (rel.end_of_support - now).days
                 return {
                     "level": rel.level,
                     "date_released": rel.date_released,
                     "description": rel.description,
                     "version": rel.version,
                     "url": rel.url,
                     "app_name": "Astronomer Runtime",
+                    "days_to_eol": days_to_eol,
```

Comment:
do we need the eol stuff for these updating warnings?

Reply:
removed it